### ISSUE TITLE
merged buffer sometimes remains in hotlist even after read

### ISSUE THREAD

**Issue by oakkitten** [weechat/weechat#215]
i have private buffers merged (`irc.look.pv_buffer=merge_all`), and sometimes, quite rarely, one of the buffers would remain unread (according to hotlist and buffers.pl). switching to other buffers/opening/closing other private buffers/typing in that buffer/receiving new messages in that buffer/cycling though buffers/zooming in and out have no effect

this bug can appear mid-conversation (i.e. surely i wasn't away), and i don't use any away scripts, and `weechat.look.hotlist_add_conditions` is default. i closed another private buffer just before it happened, but can't say if i closed anything the last time it happened or not

using WeeChat 1.1-dev [compiled on Sep 17 2014 18:04:42] but the bug appeared on some earlier build from maybe months ago, can't tell which

`/script list` if you want: beep.pl 1.1 buffers.pl 4.8  colorize_lines.pl 3.2 iset.pl 3.5 jump_smart_higher.pl 0.1 sysinfo.pl 0.7 grep.py 0.7.3  prism.py 0.2.7 TinyUrl.py 3.8

upd. the buffer would become read after some time eventually, again, no telling what made the problem disappear

**Comment by oakkitten** [comment]
finally a way to reproduce. do `$ weechat -d /tmp/abc` and connect, then

    /q foo
    /q bar
    /buffer 2 (focus foo)
    /buffer merge 3
    alt-x (/input zoom_merged_buffer, to zoom in)
    receive a message on bar
    /buffer unmerge (still being on buffer 2)

now buffer 3 (bar) doesn't get cleared when you focus it and can only be cleared with alt-h

**Comment by flashcode** [comment]
Thanks for the way to reproduce, I'll check that later, for now I just reopen the issue.

**Comment by liskin** [comment]
I think this one is still reproducible. I was just about to submit a new bug labelled "inactive zoomed merged buffers not marked as read by weechat-android" but I believe it might very well just be this one, so I'll post my reproducer here.

A little background on my setup: I heavily use merged buffers, so that Alt-2 gives me work chats, Alt-3 high priority low-traffic friends chats, and so on up to Alt-7 with general noise chanells and Alt-8 to Alt-0 queries with wife etc. And I almost exclusively zoom those merged buffers as that makes it easier to follow conversations and less likely to send a message to the wrong channel/person. Alt-a and go.py help a lot to. So I do suffer from all of #764, #1005, #562, but it's tolerable as the layout changes very rarely.

But this one is more annoying as I run into it daily. Whenever I open a buffer from weechat-android that doesn't happen to be the active (top) one from the zoomed merged group, its read status is not updated and it stays in hotlist. But that's just the console UI, its read status _is_ updated in weechat-android, even after reconnecting.

Well, and the read marker sometimes moves and sometimes doesn't, I haven't found a pattern in that yet. :-/

(@sim642 I've seen that you have a few pull requests open related to zoomed buffers. Is your setup in any way similar to mine? Do you have a branch with all your fixes that you're running locally that I can test so that if I ever get annoyed enough to attempt fixing it myself, we don't duplicate efforts?)

**Comment by sim642** [comment]
@liskin No, my setup isn't like yours: I don't use merged buffers like that nor do I really use the hotlist. Also, I don't maintain a branch with all my PRs. If you want to try them, just merge the few you want locally yourself, you'll have to compile yourself either way. Although I doubt my PRs would do anything about this issue or whatever you're experiencing.

**Comment by liskin** [comment]
@sim642 Ok, thanks for the quick answer.
